overlord/devicestate/devicestate.go

```diff
@@ -488,26 +524,36 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, pathSI *pat
 		return nil, err
 	}
 
+	revOpts := revisionOptionsForRemodel(newModelSnapChannel, ms.newRequiredRevision, ms.newModelValidationSets)
+
 	if ms.currentSnap == ms.newSnap {
 		// new model uses the same base, kernel or gadget snap
-		changed := false
+		channelChanged := false
 		if ms.new.Grade() != asserts.ModelGradeUnset {
 			// UC20 models can specify default channel for all snaps
 			// including base, kernel and gadget
-			changed, err = installedSnapChannelChanged(st, ms.newSnap, newModelSnapChannel)
+			channelChanged, err = installedSnapChannelChanged(st, ms.newSnap, newModelSnapChannel)
 			if err != nil {
 				return nil, err
 			}
 		} else if ms.canHaveUC18PinnedTrack() {
 			// UC18 models could only specify track for the kernel
 			// and gadget snaps
-			changed = ms.currentModelSnap.PinnedTrack != ms.newModelSnap.PinnedTrack
+			channelChanged = ms.currentModelSnap.PinnedTrack != ms.newModelSnap.PinnedTrack
+		}
+
+		newRevision, err := installedSnapRevisionChanged(st, ms.newSnap, ms.newRequiredRevision)
```

Comment:
Nitpick: for consistency (with other changes below)
```suggestion
		revisionChanged, err := installedSnapRevisionChanged(st, ms.newSnap, ms.newRequiredRevision)
```